docs/blog/2021-09-07-mocking-a-production-api.md

```diff
@@ -0,0 +1,300 @@
+could get. We wanted to prototype a solution over (roughly) 5 days as a side
+project - this blog post is an overview of that journey.
+
+## Day 1: Picking an API
+
+So much to do - so little time. We had decided we wanted to mock a popular API
+but didn't know where to start. We knew companies like Stripe have an
+excellent [testmode](TODO]) and even an open source http server which you can
+use instead.
+
+We decided to ask the internet 'Which API have you been struggling to test
+against' on various forums like Reddit, HN and others. Ok so now we wait and see
+let's see what the internet has to say.
+
+## Day 2: Choosing Shopify
+
+Surprisingly (or perhaps not) the internet responded. A bunch of people
+responded and primarily complained about payment processors (except for Stripe
+which was explicitly praised yet again!). A few products and companies came up
+repeatedly as being a nightmare to test against. We qualitatively evaluated the
+internet's feedback and reviewed documentation from the different APIs mentioned
+for ease of implementation (we had 3.5 days left so we couldn't pick anything
+too big). In the end we decided to go with the Shopify API!
+
+// TODO write more here about listening to user feedback
+
+Now the Shopify API is pretty big - and we're building a mock server POC from
+scratch, so again we chose to narrow down and try to mock a single endpoint
+first. We ended up picking the [Events API](TODO) which seemed pretty straight
+forward. Looking at the Events API there are three dimensions to consider when
+designing our POC solution.
+
+### 1. The Data Model
+
+The Events API returns a JSON payload which is a collection of Events. An
+example Event can be seen below:
+
+```json
+{
+    "arguments": "Ipod Nano - 8GB",
+    // Refers to a certain event and its resources.
+    "body": null,
+    // A text field containing information about the event.,
+    "created_at": "2015-04-20T08:33:57-11:00",
+    // The date and time (ISO 8601 format) when the event was created.
+    "id": 164748010,
+    // The ID of the event.
+    "desciption": "Received a new order",
+    // A human readable description of the event.
+    "path": "/admin/orders/406514653/transactions/#1145",
+    // A relative URL to the resource the event is for, if applicable.
+    "message": "Received a new order",
+    // A human readable description of the event. Can contain some HTML formatting.
+    "subject_id": 406514653,
+    // The ID of the resource that generated the event.
+    "subject_type": "Order",
+    // The type of the resource that generated the event.
+    "verb": "confirmed"
+    // The type of event that occurred. 
+}
+```
+
+Off the bat it's clear that there is some business logic that needs to be
+implemented. For example, there is some notion of causality, i.e. an `Order`
+cannot be `closed` before its been `placed`. This non-trivial business logic was
+good news - it means we can piggy back off some of the complex data generation
```

Comment:
I don't understand this sentence. Because it is non-trivial you can piggy back off synth?